.travis.yml

```diff
@@ -20,9 +20,10 @@ before-install:
   - psql -U postgres -c "ALTER USER postgres WITH PASSWORD 'password';"
 
 script:
-  - bin/ameba
-  - crystal tool format --check
-  - crystal spec $TEST_SUITE
+  - docker build -t amber-test .
```

Comment:
consider using:
```
docker-compose run spec
```

Reply:
Ah, good callout.

Reply:
I attempted to use docker-compose here, but received the following errors when running tests:

```bash
Failures:

  1) database sqlite test creates the database when db migrate
     Failure/Error: File.exists?(db_filename).should be_true

       Expected: true
            got: false

     # spec/amber/cli/commands/database_spec.cr:35

  2) Amber::Pipe::Session session persist across different requests Redis Store sets session value in controller

       Socket::ConnectError: Error connecting to 'localhost:6379': Cannot assign requested address (Redis::CannotConnectError)
         from lib/redis/src/redis/socket_wrapper.cr:12:5 in 'initialize'
         from lib/redis/src/redis/connection.cr:8:3 in 'new'
         from lib/redis/src/redis.cr:157:19 in 'connect'
         from lib/redis/src/redis.cr:117:5 in 'initialize:url'
         from lib/redis/src/redis.cr:95:3 in 'new:url'
         from src/amber/router/session/session_store.cr:19:7 in 'redis_store'
         from src/amber/router/session/session_store.cr:10:31 in 'build'
         from src/amber/router/context.cr:30:18 in 'session'
         from spec/amber/pipes/session_spec.cr:41:13 in '->'
         from ../../usr/share/crystal/src/primitives.cr:255:3 in 'internal_run'
         from ../../usr/share/crystal/src/spec/example.cr:33:16 in 'run'
         from ../../usr/share/crystal/src/spec/context.cr:18:23 in 'internal_run'
         from ../../usr/share/crystal/src/spec/context.cr:330:7 in 'run'
         from ../../usr/share/crystal/src/spec/context.cr:18:23 in 'internal_run'
         from ../../usr/share/crystal/src/spec/context.cr:330:7 in 'run'
         from ../../usr/share/crystal/src/spec/context.cr:18:23 in 'internal_run'
         from ../../usr/share/crystal/src/spec/context.cr:330:7 in 'run'
         from ../../usr/share/crystal/src/spec/context.cr:18:23 in 'internal_run'
         from ../../usr/share/crystal/src/spec/context.cr:147:7 in 'run'
         from ../../usr/share/crystal/src/spec/dsl.cr:270:7 in '->'
         from ../../usr/share/crystal/src/primitives.cr:255:3 in 'run'
         from ../../usr/share/crystal/src/crystal/main.cr:45:14 in 'main'
         from ../../usr/share/crystal/src/crystal/main.cr:114:3 in 'main'
         from __libc_start_main
         from _start
         from ???
```

It looks like I would have had to modify tests and come configs. I will look into docker-compose in another iteration, I was mainly concerned with having the proper version of crystal run in CI at the moment.

Reply:
Ok, this is the `REDIS_URL` and `DATABASE_URL`.  I think we can support environment variables here.